honeybee_vtk/config.py

```diff
@@ -36,10 +183,211 @@ def validate_object(cls, v: str) -> str:
 
     @validator('file_paths')
     def validate_paths(cls, v: List[str]) -> List[str]:
-        if all([Path(path).is_file() for path in v]):
+        if all([pathlib.Path(path).is_file() for path in v]):
             return v
         else:
-            paths = tuple([path for path in v if not Path(path).is_file()])
+            paths = tuple([path for path in v if not pathlib.Path(path).is_file()])
             raise ValueError(
                 f'Following file paths are not valid {paths}.'
             )
+
+
+def _validate_data(data: DataConfig, model: Model) -> bool:
+    """Cross check data with model.
+
+    For grids, it will be checked if the number of data files and the names of the
+    data files match with the grid identifiers. For other than grid objects, it will
+    be checked that only one data file is provided and the length of the data file
+    matches the length of Polydata in the model. This is a helper method to the
+    public load_config method.
+
+    Args:
+        data: A DataConfig object.
+        model: A honeybee-vtk model.
+
+    Returns:
+        A boolean value.
+    """
+    # if file_paths is empty
+    if not data.file_paths:
+        raise ValueError(
+            f'For object with name {data.name} There are not file paths'
+            ' provided to load data from.'
+        )
+
+    # if object name is "grid" check that the name of files match the grid names
+    if data.object_type == model_dataset_names[-1]:
+
+        # make sure grids are loaded on the model if grid data is to be mounted
+        assert len(model.sensor_grids.data) > 0, 'Sensor grids are not loaded on'\
+            ' this model. Reload them using grid options.'
+
+        grid_names = [grid.identifier for grid in model.sensor_grids.data]
+        file_names = [pathlib.Path(path).stem for path in data.file_paths]
+        names_in_grids = all([name in grid_names for name in file_names])
+
+        # the number of data files must not exceed the number of grids
+        if len(file_names) > len(grid_names):
+            raise ValueError(
+                f'There are only {len(grid_names)} grids in the model and the'
+                f' identifiers of those grids are {tuple(grid_names)}.'
+            )
+
+        # TODO: remove this check in the next iteration
+        # make sure there's one file for each grid in the model
+        if not names_in_grids:
+            raise ValueError(
+                'Make sure the file names match the grid identifiers. The'
+                f' identifiers of the grids in the model are {tuple(grid_names)}.'
+            )
+
+        # make sure length of each file matches the number of sensors in grid
+        file_lengths = [get_line_count(file_path) for file_path in data.file_paths]
```

Comment:
This check can become very costly depending on the file size but for now I think it should be fine.

Reply:
I captured this in an [issue](https://github.com/ladybug-tools/honeybee-vtk/issues/168).